Shoko.Server/Server/StartServer.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using Shoko.Server.Settings;
+using NLog;
+namespace Shoko.Server.Server;
+#nullable enable
+public static class StartServer
+{
+    private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+    public static void StartupServer(Action? addEventHandlers, Func<bool> serverStart)
+    {
+        SetInstanceIfNeeded();
+        InitInstanceLogger();
+        LoadSettings();
+        if (serverStart?.Invoke() == false)
+            return;
+        EnsureAniDBSocketInitialized();
+        addEventHandlers?.Invoke();
+    }
+
+    public static string? GetInstanceFromCommandLineArguments()
+    {
+        const int notFound = -1;
+        var       args     = Environment.GetCommandLineArgs();
+        var       idx      = Array.FindIndex(args, x => string.Equals(x, "instance", StringComparison.InvariantCultureIgnoreCase));
+        if (idx is notFound)
+            return null;
+        if (idx >= args.Length - 1)
+            return null;
+        return args[idx + 1];
+    }
```

Comment:
Given the fact that shoko's "instance" support is a really niche thing, I think this is over-engineering a solution for finding it, simply a `if argv[1] == 'instance' return argv[2]` would be fine.

Reply:
having a simple solution such as if[1] then [2] is prone to breaking and wont allow any further development of command line arguments, without rewriting. i also think such a solution would be too hacky, since it does not respect array bounds, and once you test on array bounds, you are really not any different from this solution.